### DISCUSSION TITLE
systemd onedrive syncs to wrong directory

### DISCUSSION THREAD

**Discussion by nwalet** [abraunegg/onedrive#1651] [Q&A] [answered]
This is a small annoyance only.
I run onedrive v2.4.13-14-g350fae1 on rocky linux 6.4 (i.e,. centos...) without any real problems with a manual sync. For historical reasons my home directory is nfs mounted on a mount point that is not /home. If I run the automated (systemd@user) client, it seems to look for the home directory (and onedrive folder) in /home, where it is not, and thus it fails.
Anyone knows how to work around this issue?

**Comment by nwalet** [comment]
Sorry, should have dug through. It is actually in the contrib/systemd/onedrive@.service* files. I can clearly fix that for my own usage (just replace /home by what I need), but need to think through a general solution...

**Comment by abraunegg** [comment, marked as answer]
@nwalet 
> I run onedrive v2.4.13-14-g350fae1 on rocky linux 6.4 (i.e,. centos...) without any real problems with a manual sync. For historical reasons my home directory is nfs mounted on a mount point that is not /home. If I run the automated (systemd@user) client, it seems to look for the home directory (and onedrive folder) in /home, where it is not, and thus it fails.
Anyone knows how to work around this issue?
> ...
> Sorry, should have dug through. It is actually in the contrib/systemd/onedrive@.service* files. I can clearly fix that for my own usage (just replace /home by what I need), but need to think through a general solution...

This is why the 'config' file exists for this type of scenario.

Your 'systemd' configuration will pick up the 'config' file from the applicable user directory - dependent on what user you are running the service as - but the configuration file will direct the client, running as a systemd service, as to what directory to use as the location for the files.

In your case, your 'config' file should specify the 'sync_dir' paramerter as pointing to your required location, eg:
```
sync_dir = "/path/to/nfs/mountpoint/OneDrive" 
```

You can also modify the systemd files directly, to include the `--syncdir ARG` parameter.

Additionally, and this has come up a *number* of times - and is *really* important - because you are mounting a non-local location:
1. In your local file system mount point, *before* you mount your NFS location, create a '.nosync' file, so that if your NFS is unmounted | not mounted - you do not loose your data on OneDrive
2. Configure in your 'config' file `check_nomount = "true"` and `check_nosync = "true"`
